pkg/controller/chaosscheduler/chaosscheduler_controller.go

```diff
@@ -0,0 +1,336 @@
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileChaosScheduler{client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("chaos-scheduler")}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("chaosscheduler-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = watchChaosResources(mgr.GetClient(), c)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// watchSecondaryResources watch's for changes in chaos resources
+func watchChaosResources(clientSet client.Client, c controller.Controller) error {
+	// Watch for Primary Chaos Resource
+	err := c.Watch(&source.Kind{Type: &schedulerV1.ChaosSchedule{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &operatorV1.ChaosEngine{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &schedulerV1.ChaosSchedule{},
+		IsController: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileChaosScheduler implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileChaosScheduler{}
+
+/*Reconcile reads that state of the cluster for a ChaosScheduler object and makes changes based on the state read
+and what is in the ChaosScheduler.Spec
+Note:
+The Controller will requeue the Request to be processed again if the returned error is non-nil or
+Result.Requeue is true, otherwise upon completion it will remove the work from the queue.*/
+func (r *ReconcileChaosScheduler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := chaosTypes.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ChaosScheduler")
+
+	// Fetch the ChaosScheduler instance
+	scheduler, err := r.getChaosSchedulerInstance(request)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	schedulerReconcile := &reconcileScheduler{
+		r:         r,
+		reqLogger: reqLogger,
+	}
+
+	switch scheduler.Instance.Spec.ScheduleState {
+	case "", schedulerV1.StateActive:
+		{
+			return schedulerReconcile.reconcileForCreationAndRunning(scheduler)
+		}
+	// case schedulerV1.StateStopped:
+	// 	{
+	// 		if !checkScheduleStatus(scheduler, schedulerV1.StatusStopped) {
+	// 			return r.reconcileForDelete(scheduler, request)
+	// 		}
+	// 	}
```

Comment:
is this not required?

Reply:
May be needed in the near future.... Not confirm whether we should have a `stop` state for the scheduler .. If yes then what will be its function

Reply:
Removed it for now